pkg/render/logstorage.go

```diff
@@ -61,20 +61,22 @@ const (
 
 	TigeraElasticsearchCertSecret = "tigera-secure-elasticsearch-cert"
 
-	ElasticsearchName                     = "tigera-secure"
+	ElasticsearchName                     = "tigera-secure-internal"
```

Comment:
ElasticsearchName

Reply:
After chatting with @rene-dekker it seems like the service name uses this name, so it makes this sort of awkward. @stevegaossou I don't think changing this name is an option, it's a very risky operation that will like result in this cluster being deleted. I think this has to go back to the drawing board, we can't change the service name because of this, so we need to figure out if that's acceptable or not to roll out to customers outside of calico cloud

Reply:
@rene-dekker had a good idea about modifying the existing service in a certain way (I'll let him explain it), I just wanted to chime in to say whatever you do can you make sure the ECK operator won't trounce over it, please review what you come up with against the service controller here https://github.com/elastic/cloud-on-k8s/blob/1.3/pkg/controller/elasticsearch/services/services.go

Reply:
@rene-dekker @Brian-McM thanks for the feedback. 

@vberezny is going to look into seeing if there is a supported way to override the service name that is generated by ECK operator (which is `tigera-secure-es-http`). 

We unfortunately need to change that name, so that es-gateway can use `tigera-secure-es-http`. This is important for Calico Cloud specifically, because we cannot require customers upgrade their managed clusters. So components need to continue using `tigera-secure-es-http`, and that service needs to be backed by es-gateway instead. 

I don't think we want to go down a path where some components (from some clusters) use `tigera-secure-es-http` and some use a new service (e.g. `tigera-secure-es-gateway-http`). 

> I think this has to go back to the drawing board, we can't change the service name because of this, so we need to figure out if that's acceptable or not to roll out to customers outside of calico cloud

I think we want to have es-gateway in place for Calico Enterprise because it has security benefits for MCM setups in general (not just Calico Cloud). But I'll leave this open and check in with Matt D and Garwood.

Reply:
Brian and I were thinking if we configure Elasticsearch (CR) with HTTP.Service.Spec to point to label k8s-app=es-gateway and create a new service with k8s-app=elasticsearch (or something like that), we can keep the existing name for the CR and hopefully everything works. Now there are two caveats:
- It only works if ECK respects that service configuration
- When requests are sent to a service with label k8s-app=elasticsearch during an ECK change between two node sets due to a spec change or whatever, I cannot predict what is going to happen.

Reply:
@vberezny I think you can override it like this: 

```
	elasticsearch := &esv1.Elasticsearch{
		TypeMeta: metav1.TypeMeta{Kind: "Elasticsearch", APIVersion: "elasticsearch.k8s.elastic.co/v1"},
		ObjectMeta: metav1.ObjectMeta{
			Name:      ElasticsearchName,
			Namespace: ElasticsearchNamespace,
			Annotations: map[string]string{
				"common.k8s.elastic.co/controller-version": components.ComponentElasticsearchOperator.Version,
			},
		},
		Spec: esv1.ElasticsearchSpec{
			Version: components.ComponentEckElasticsearch.Version,
			Image:   es.esImage,
			HTTP: cmnv1.HTTPConfig{
				TLS: cmnv1.TLSOptions{
					Certificate: cmnv1.SecretRef{
						SecretName: TigeraElasticsearchCertSecret,
					},
				},
                                 // -------------------------- ADD THE BELOW--------------------------
				Service: cmnv1.ServiceTemplate{
					ObjectMeta: metav1.ObjectMeta{
						Name: "tigera-secure-internal-es-http",
					},
				},
				// ----------------------------------------------------------------------
			},
			NodeSets: es.nodeSets(),
		},
	}

```

Reply:
ObjectMeta cannot be customized according to the ECK docs. Only the spec.

Reply:
That's sad. 😞 

Vlad mentioned the next option you guys are trying (of using ECK operator to setup the service to be backed by ES gateway pods) and then create the other HTTP service outside of ECK operator and have that be backed by ES pods. 

A problem might be how Kibana pods are configured to talk to ES pods (which ECK operator configures.

Reply:
Kibana shouldn't have any problem @stevegaossou, Kibana just needs the url and the certificate for Elastcisearch, and those won't change.